src/gp_instance.cpp

```diff
@@ -174,3 +175,63 @@ void GPInstance::EstimateSBNParameters() {
   double marginal_log_lik = GetEngine()->GetLogMarginalLikelihood();
   std::cout << std::setprecision(9) << marginal_log_lik << std::endl;
 }
+
+RootedTreeCollection GPInstance::GetRootedTreeCollection() {
+
+  Tree::TreeVector tree_vector;
+  Node::NodePtrVec trees = dag_.GenerateAllTrees();
+  EigenVectorXd bl = engine_->GetBranchLengths();
```

Comment:
const ref?

Reply:
Using `EigenVectorXdRef`fails to compile. Not sure why.